diff --git a/src/dxvk/dxvk_surface.cpp b/src/dxvk/dxvk_surface.cpp
index 20dc456..3420de3 100644
--- a/src/dxvk/dxvk_surface.cpp
+++ b/src/dxvk/dxvk_surface.cpp
@@ -88,11 +88,11 @@ namespace dxvk {
   uint32_t DxvkSurface::pickImageCount(
     const VkSurfaceCapabilitiesKHR& caps,
           VkPresentModeKHR          mode) const {
-    uint32_t count = caps.minImageCount;
+    uint32_t count = caps.minImageCount + 1;
     
-    if (mode == VK_PRESENT_MODE_MAILBOX_KHR
-     || mode == VK_PRESENT_MODE_FIFO_KHR)
-      count += 1;
+    // if (mode == VK_PRESENT_MODE_MAILBOX_KHR
+    //  || mode == VK_PRESENT_MODE_FIFO_KHR)
+    //   count += 1;
     
     if (count > caps.maxImageCount && caps.maxImageCount != 0)
       count = caps.maxImageCount;
