diff --git a/src/dxvk/dxvk_swapchain.cpp b/src/dxvk/dxvk_swapchain.cpp
index 3657334..bf7445c 100644
--- a/src/dxvk/dxvk_swapchain.cpp
+++ b/src/dxvk/dxvk_swapchain.cpp
@@ -140,7 +140,7 @@ namespace dxvk {
     auto swapImages = this->retrieveSwapImages();
     
     m_framebuffers.resize(swapImages.size());
-    m_semaphoreSet.resize(swapImages.size());
+    m_semaphoreSet.resize(swapImages.size() * 2);
     
     DxvkImageCreateInfo imageInfo;
     imageInfo.type          = VK_IMAGE_TYPE_2D;
@@ -172,7 +172,9 @@ namespace dxvk {
     for (size_t i = 0; i < swapImages.size(); i++) {
       m_framebuffers.at(i) = m_device->createImageView(
         new DxvkImage(m_vkd, imageInfo, swapImages.at(i)), viewInfo);
-      
+    }
+
+    for (size_t i = 0; i < m_semaphoreSet.size(); i++) {
       m_semaphoreSet.at(i).acquireSync = m_device->createSemaphore();
       m_semaphoreSet.at(i).presentSync = m_device->createSemaphore();
     }
